app/services/katello/repository_type_manager.rb

```diff
@@ -1,35 +1,78 @@
 module Katello
   class RepositoryTypeManager
-    @repository_types = {}
+    PULP3_FEATURE = "Pulpcore".freeze
+
+    @defined_repository_types = {}
+    @usable_repository_types = {}
+    begin
+      @pulp_primary = ::SmartProxy.unscoped.detect { |proxy| !proxy.setting(PULP3_FEATURE, 'mirror') }
+    rescue ActiveRecord::StatementInvalid
+      @pulp_primary = nil
+    end
     class << self
       private :new
-      attr_reader :repository_types
+      attr_reader :defined_repository_types
 
       # Plugin constructor
       def register(id, &block)
-        configured = SETTINGS[:katello][:content_types].nil? || SETTINGS[:katello][:content_types].with_indifferent_access[id]
-        if find(id).blank? && configured
-          repository_type = ::Katello::RepositoryType.new(id)
-          repository_type.instance_eval(&block) if block_given?
-          repository_types[id.to_s] = repository_type
+        if @pulp_primary&.has_feature?(PULP3_FEATURE) && @pulp_primary&.capabilities(PULP3_FEATURE)&.empty?
+          @pulp_primary&.refresh
+          if @pulp_primary&.capabilities(PULP3_FEATURE).&empty?
+            fail Katello::Errors::PulpcoreMissingCapabilities
+          end
+        end
+        defined_repo_type = find_defined(id)
+        if defined_repo_type.blank?
+          defined_repo_type = ::Katello::RepositoryType.new(id)
+          defined_repo_type.instance_eval(&block) if block_given?
+          defined_repository_types[id.to_s] = defined_repo_type
+        end
+        if find(id).blank? && defined_repo_type.present?
+          usable_repository_types[id.to_s] = defined_repo_type if pulp3_plugin_installed?(id)
         end
       end
 
+      def usable_repository_types
+        if (defined_repository_types.keys - @usable_repository_types.keys).present?
+          defined_repository_types.keys.each { |repo_type| update_usable_repository_type(repo_type.to_s) }
+        end
+        @usable_repository_types
+      end
+
       def creatable_repository_types
-        repository_types.select do |repo_type, _|
+        usable_repository_types.select do |repo_type, _|
           creatable_by_user?(repo_type)
         end
       end
 
+      def repo_type_to_pulp3_plugin(repo_type)
+        case repo_type
+        when ::Katello::Repository::YUM_TYPE
+          'rpm'
+        when ::Katello::Repository::DOCKER_TYPE
+          'container'
+        when ::Katello::Repository::FILE_TYPE
+          'file'
+        when ::Katello::Repository::ANSIBLE_COLLECTION_TYPE
+          'ansible'
+        when ::Katello::Repository::DEB_TYPE
+          'deb'
+        end
+      end
```

Comment:
I think this mapping may already exist indirectly 

```
Katello::RepositoryTypeManager.repository_types['docker'].pulp3_plugin
```

Reply:
The issue is that I need this mapping before the repository types are registered.

Reply:
i dug through the code and it looks like this is called during the repository_type registration, and its actually registered by this point. I tested this change and it worked fine, maybe it didn't on a previous incarnation?